CONTRIBUTING.rst

```diff
@@ -2,38 +2,56 @@
 Contributing to Klein
 =====================
 
-Klein is an open source project that welcomes contributions of all kinds coming from the community.
-
-These include:
+Klein is an open source project that welcomes contributions of all kinds coming from the community, including:
 
 - code patches,
 - `documentation <http://klein.readthedocs.org/>`_ improvements,
 - `bug reports <https://github.com/twisted/klein/issues>`_,
 - reviews for `contributed patches <https://github.com/twisted/klein/pulls>`_.
 
-Here's some suggestions to make the contributing process much smoother for you:
+
+Getting started
+===============
+
+Here is a list of shell commands that will install the dependencies of Klein, run the test suite, and compile the documentation.
+
+::
+   pip install --user -r requirements.txt
+   tox
+   tox -e docs
+
+
+Next steps
+----------
+
+Here are some suggestions to make the contributing process easier for everyone:
 
 Code
-====
+----
 
 - Propose all patches through a pull request in the `GitHub repo <https://github.com/twisted/klein>`_.
-- Use `Twisted's code style guide <http://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html>`_ for all code you touch, as long as it doesn't break a public API.
```

Comment:
The "as long as it doesn't break a public API" is important here -- some parts of Klein (eg. `klein.resource.ensure_utf8_bytes`) do not adhere to Twisted's style guide, and so changing it to bring it in line will break the compatibility guarantees of a Twisted project. We don't actually mention a compatibility policy here, so maybe we should link https://twistedmatrix.com/trac/wiki/CompatibilityPolicy to be explicit, rather than have it just be implicit as it has been currently.